pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,42 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		// IF the VMI has a pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod == nil {
+				continue
+			}
+			for i, iface := range vmiCopy.Status.Interfaces {
+				if iface.Name == network.Name {
+					if vmiCopy.Status.Interfaces[i].IP != pod.Status.PodIP {
+						vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+						networkPatchNeeded = true
+
+						for j, ip := range iface.IPs {
+							if ip == iface.IP {
+								vmiCopy.Status.Interfaces[i].IPs[j] = pod.Status.PodIP
+							}
+						}
+					}
+				}
+			}
+			// Currently will only traverse to the first Pod network found
```

Comment:
we are ok with it ? might be an errata in the future ?

Reply:
my understanding with multiple pod networks is limited here, @oshoval can you describe more here?

Reply:
tbh mine as well, 
thought just generally what happens if there are more than one pod network, 
do we need to support this feature across all of them ?